frontend/src/components/common/Header/Header.tsx

```diff
@@ -1,60 +1,84 @@
-import React, { useState, ReactElement } from 'react';
-import { useDispatch } from 'react-redux';
+import React, { useState, useEffect, ReactElement } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link, useNavigate, useSearchParams } from 'react-router-dom';
 import classNames from 'classnames';
 import { FaSearch } from 'react-icons/fa';
 import { FiShoppingBag } from 'react-icons/fi';
-import { NavLink } from 'react-router-dom';
-import { updateName, updateUrl, resetAll } from '../../../pages/Home/components/Criteria/filterSlice';
-import handleInputEvent from '../../../assets/helper/handleInputEvent';
+import { CgClose } from 'react-icons/cg';
+import parseToSearchUrl from '@assets/helper/parseToSearchUrl';
+import handleInputEvent from '@assets/helper/handleInputEvent';
+import parseFilterURLParams from '@assets/helper/parseFilterURLParam';
+import { updateName, updateUrl, updateAll, resetAll, updateCategory } from '@pages/Home/Criteria/filterSlice';
+import rootState from '@models/rootState';
 import './Header.scss';
+import Cart from './Cart/Cart';
 
-const Header: React.FC = (): ReactElement => {
-  const navLinks = [
-    {
-      title: 'All',
-      slug: '/search/all',
-    },
-    {
-      title: 'New Arrival',
-      slug: '/search/new-arrivals',
-    },
-    {
-      title: 'Featured',
-      slug: '/search/featured',
-    },
-  ];
+const navLinks = [
+  {
+    title: 'All',
+    slug: '/search/',
+  },
+  {
+    title: 'New Arrivals',
+    slug: '/search/filter?category=New+Arrivals',
+  },
+  {
+    title: 'Featured',
+    slug: '/search/filter?category=Featured',
+  },
+];
+const Header = (): ReactElement => {
   const [search, setSearch] = useState('');
-
+  const [openCart, setOpenCart] = useState(false);
+  const [selected, setSelected] = useState(-1);
+  const [searchParams] = useSearchParams();
+  const searchParamsObject = parseFilterURLParams(searchParams);
+  const searchParamsString = parseToSearchUrl(searchParamsObject);
+  const navigate = useNavigate();
   const dispatch = useDispatch();
+  const cartList = useSelector((state: rootState) => state.cart.cartList);
+  const isEmptyCart = cartList.length === 0;
+  useEffect(() => {
+    const idx = navLinks.findIndex((item) => item.slug.includes(searchParamsString));
+    setSelected(idx);
+    dispatch(updateAll(searchParamsObject));
+    dispatch(updateUrl());
+  }, [searchParams]);
   const handleSearch = (): void => {
     dispatch(updateName(search));
     dispatch(updateUrl());
   };
-  const handleNavBar = (title: string): void => {
+
+  const handleNavBar = (title: string, idx: number): void => {
+    navigate(navLinks[idx].slug);
     if (title === navLinks[0].title) {
```

Comment:
You shouldn't use the title for checking, create a new id and use it instead. And don't use sth like `navLinks[0]`, if sbd changes the position, your code will break, use `navLinks.filter(({id}) => id === NAV_LINK.ALL)` instead

Reply:
@thuanvuong-kms `filter `will return a array of suitable item. so that the condition will be like 
`if(title === navLinks.filter((item) => item.title === NAV_LINK.ALL)[0].title)`. Is it acceptable ?